baselines/pfsp/pfsp_gpu_cuda.c

```diff
@@ -0,0 +1,536 @@
+  for (int i = 0; i < jobs; i++) {
+    root->prmu[i] = i;
+  }
+}
+
+/*******************************************************************************
+Implementation of a dynamic-sized single pool data structure.
+Its initial capacity is 1024, and we reallocate a new container with double
+the capacity when it is full. Since we perform only DFS, it only supports
+'pushBack' and 'popBack' operations.
+*******************************************************************************/
+
+// Pools are managed by the CPU only
+
+#define CAPACITY 1024
+
+typedef struct
+{
+  Node* elements;
+  int capacity;
+  int size;
+} SinglePool;
+
+void initSinglePool(SinglePool* pool)
+{
+  pool->elements = (Node*)malloc(CAPACITY * sizeof(Node));
+  pool->capacity = CAPACITY;
+  pool->size = 0;
+}
+
+void pushBack(SinglePool* pool, Node node)
+{
+  if (pool->size >= pool->capacity) {
+    pool->capacity *= 2;
+    pool->elements = (Node*)realloc(pool->elements, pool->capacity * sizeof(Node));
+  }
+
+  pool->elements[pool->size++] = node;
+}
+
+Node popBack(SinglePool* pool, int* hasWork)
+{
+  if (pool->size > 0) {
+    *hasWork = 1;
+    return pool->elements[--pool->size];
+  }
+
+  return (Node){0};
+}
+
+void deleteSinglePool(SinglePool* pool)
+{
+  free(pool->elements);
+}
+
+/*******************************************************************************
+Implementation of the parallel CUDA GPU PFSP search.
+*******************************************************************************/
+
+void parse_parameters(int argc, char* argv[], int* inst, int* lb, int* ub, int* m, int *M)
+{
+  *m = 25;
+  *M = 50000;
+  *inst = 14;
+  *lb = 1;
+  *ub = 1;
+  /*
+    NOTE: Only forward branching is considered because other strategies increase a
+    lot the implementation complexity and do not add much contribution.
+  */
+
+  // Define long options
+  static struct option long_options[] = {
+    {"inst", required_argument, NULL, 'i'},
+    {"lb", required_argument, NULL, 'l'},
+    {"ub", required_argument, NULL, 'u'},
+    {"m", required_argument, NULL, 'm'},
+    {"M", required_argument, NULL, 'M'},
+    {NULL, 0, NULL, 0} // Terminate options array
+  };
+
+  int opt, value;
+  int option_index = 0;
+
+  while ((opt = getopt_long(argc, argv, "i:l:u:m:M:", long_options, &option_index)) != -1) {
+    value = atoi(optarg);
+
+    switch (opt) {
+    case 'i':
+      if (value < 1 || value > 120) {
+	fprintf(stderr, "Error: unsupported Taillard's instance\n");
+	exit(EXIT_FAILURE);
+      }
+      *inst = value;
+      break;
+
+    case 'l':
+      if (value < 0 || value > 2) {
+	fprintf(stderr, "Error: unsupported lower bound function\n");
+	exit(EXIT_FAILURE);
+      }
+      *lb = value;
+      break;
+
+    case 'u':
+      if (value != 0 && value != 1) {
+	fprintf(stderr, "Error: unsupported upper bound initialization\n");
+	exit(EXIT_FAILURE);
+      }
+      *ub = value;
+      break;
+
+    case 'm':
+      if (value < 1 || value > 100) {
```

Comment:
Please remove also `|| value > 100`. Setting large values of `m` should be allowed, even if it's not necessarily efficient